src/sdl2_compat.c

```diff
@@ -3914,6 +3914,231 @@ DECLSPEC void SDLCALL SDL_SIMDFree(void *ptr)
+    case AUDIO_F32MSB:
+        return SDL_TRUE; /* supported. */
+
+    default:
+        break;
+    }
+
+    return SDL_FALSE; /* unsupported. */
+}
+
+static SDL_bool SDL_IsSupportedChannelCount(const int channels)
+{
+    return ((channels >= 1) && (channels <= 8)) ? SDL_TRUE : SDL_FALSE;
+}
+
+
+typedef struct {
+    SDL_AudioFormat src_format;
+    Uint8 src_channels;
+    int src_rate;
+    SDL_AudioFormat dst_format;
+    Uint8 dst_channels;
+    int dst_rate;
+} AudioParam;
+
+
+DECLSPEC int SDLCALL SDL_BuildAudioCVT(SDL_AudioCVT *cvt,
+                                       SDL_AudioFormat src_format,
+                                       Uint8 src_channels,
+                                       int src_rate,
+                                       SDL_AudioFormat dst_format,
+                                       Uint8 dst_channels,
+                                       int dst_rate)
+{
+    SDL_AudioStream *stream;
+
+    /* Sanity check target pointer */
+    if (cvt == NULL) {
+        return SDL_InvalidParamError("cvt");
+    }
+
+    /* Make sure we zero out the audio conversion before error checking */
+    SDL_zerop(cvt);
+
+#if 0
+    if (!SDL_IsSupportedAudioFormat(src_format)) {
+        return SDL_SetError("Invalid source format");
+    }
+    if (!SDL_IsSupportedAudioFormat(dst_format)) {
+        return SDL_SetError("Invalid destination format");
+    }
+    if (!SDL_IsSupportedChannelCount(src_channels)) {
+        return SDL_SetError("Invalid source channels");
+    }
+    if (!SDL_IsSupportedChannelCount(dst_channels)) {
+        return SDL_SetError("Invalid destination channels");
+    }
+    if (src_rate <= 0) {
+        return SDL_SetError("Source rate is equal to or less than zero");
+    }
+    if (dst_rate <= 0) {
+        return SDL_SetError("Destination rate is equal to or less than zero");
+    }
+    if (src_rate >= SDL_MAX_SINT32 / RESAMPLER_SAMPLES_PER_ZERO_CROSSING) {
+        return SDL_SetError("Source rate is too high");
+    }
+    if (dst_rate >= SDL_MAX_SINT32 / RESAMPLER_SAMPLES_PER_ZERO_CROSSING) {
+        return SDL_SetError("Destination rate is too high");
+    }
+#endif
+
+    stream = SDL3_CreateAudioStream(src_format, src_channels, src_rate,
+            dst_format, dst_channels, dst_rate);
+    if (stream == NULL) {
+        return -1;
+    }
+
+#if DEBUG_CONVERT
+    SDL_Log("SDL_AUDIO_CONVERT: Build format %04x->%04x, channels %u->%u, rate %d->%d\n",
+            src_format, dst_format, src_channels, dst_channels, src_rate, dst_rate);
+#endif
+
+    /* Start off with no conversion necessary */
+    cvt->src_format = src_format;
+    cvt->dst_format = dst_format;
+    cvt->needed = 0;
+    cvt->filter_index = 0;
+    SDL_zeroa(cvt->filters);
+    cvt->len_mult = 1;
+    cvt->len_ratio = 1.0;
+    cvt->rate_incr = ((double)dst_rate) / ((double)src_rate);
+
+    /* Use the filters[] to store some data ... */
+    {
+        AudioParam *ptr = (AudioParam *)&cvt->filters[1];
```

Comment:
Let's move this to the very last possible position in the array, so there's less chance that an app will overwrite it if it tries to do something dangerous like add its own functions to the end of the filter list.

(SDL2 did this, too, in SDL_ResampleCVT, for the same reason.)

Reply:
yes done!